app/assets/javascripts/components/buttons/generic-button-component.js

```diff
@@ -0,0 +1,41 @@
+ManageIQ.angular.app.component('genericButtonComponent', {
+  bindings: {
+    angularForm: '=',
+    modelCopy: '=',
+    model: '=',
+    redirectUrl: '=?',
+    newRecord: '<',
+    entity: '=?',
+    entityName: '=?',
+    onSaveClicked: '&?',
+    onAddClicked: '&?',
+  },
+  controllerAs: 'vm',
+  controller: genericButtonController,
+  templateUrl: '/static/buttons/generic_button_component.html.haml',
+});
+
+genericButtonController.$inject = ['miqService', '$window'];
+
+function genericButtonController(miqService) {
+  var vm = this;
+
+  vm.$onInit = function() {
+    vm.saveable = miqService.saveable;
+
+    vm.resetClicked = function() {
```

Comment:
Well.. you'll find out eventually that if you want to remove redundancy, you should not be handling the same thing in two different places :).

Because guess what, you're introducing exactly the kind of redundancy we should be getting rid of here...

   * when Cancel is clicked, the buttons component will do flash message, redirect, etc.
   * when Save is clicked, the *form* will do flash messages, redirect, etc.

You yourself can see that moving `Save` to the buttons component is not practical, would require too much knowloedge of the form. But then we do the same thing in both the form and the buttons. How does that hepl redundancy?

I'm all for reducing that redundancy, and if I wasn't busy with webpack now, I'd show you how I mean to do that, but .. no, this won't help a bit.

(Also, as I said again and again, we should be reducing the number of button partials, not increasing them, so I would really like to see a component that you can use in all the places first, before we move forward with any further componentization of forms.)